phantazm-zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/GunStats.java

```diff
@@ -0,0 +1,95 @@
+package com.github.phantazmnetwork.zombies.equipment.gun;
+
+import com.github.phantazmnetwork.commons.Namespaces;
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.collection.ConfigNode;
+import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.key.Keyed;
+import org.jetbrains.annotations.NotNull;
+
+public record GunStats(long shootSpeed,
+                       long reloadSpeed,
+                       int maxAmmo,
+                       int maxClip,
+                       int shots,
+                       long shotInterval) implements Keyed {
+
+    public static final Key SERIAL_KEY = Key.key(Namespaces.PHANTAZM, "gun.stats");
+
+    public static @NotNull ConfigProcessor<GunStats> processor() {
+        return new ConfigProcessor<>() {
+
+            @Override
+            public @NotNull GunStats dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+                long shootSpeed = element.getNumberOrThrow("shootSpeed").longValue();
+                if (shootSpeed < 0) {
```

Comment:
would be nice to have a private static util method like throwIfTrue(boolean condition, String message) throws ConfigProcessException

Reply:
also, in general, we haven't been validating data to see if it's in reasonable ranges -- if we do it here we should do it everywhere for consistency, but doing it everywhere will be tedious and add a lot of boilerplate

Reply:
we probably should -- also I did kind of validate both in data constructor and in config processor, is there a better way lol